src/app/components/dashboard-company-deletion/dashboard-company-deletion.coffee

```diff
@@ -0,0 +1,21 @@
+angular.module 'mnoEnterpriseAngular'
+  .component('dashboardCompanyDeletion', {
+    restrict: 'EA'
+    templateUrl: 'app/components/dashboard-company-deletion/dashboard-company-deletion.html',
+    controllerAs: "DashboardCompanyDeletionCtrl"
+    controller:  ($uibModal, MnoeCurrentUser, MnoeOrganizations, MnoeAppInstances, MnoConfirm) ->
+      vm = this
+      vm.isDeletionOpen = false
+
+      vm.proceed = (reason, password) ->
+        MnoeOrganizations.freeze(reason, password)
+
+      vm.requestDeletion =  () ->
+        modalInstance = $uibModal.open(
+          component: "dashboardDeletionConfirm",
+          size: "lg",
+          resolve:
+            actionCb: () ->  vm.proceed
```

Comment:
I think you can write `actionCb: vm.proceed` (Remove the closure)

Reply:
Without it there is an error: explicit annotation required. The [docs](https://angular-ui.github.io/bootstrap/#ui-router-resolves) uses functions for each `resolve` property. I am not sure how it works, but I can't seem to make it work without a closure